src/services/userService.js

```diff
@@ -0,0 +1,45 @@
+let users = [];
+
+const resetUsers = () => {
+  users = [];
+};
+
+const getAllUsers = () => {
+  return users;
+};
+
+const getUserById = (id) => {
+  return users.find((person) => person.id === +id) || null;
+};
+
+const createUser = (name) => {
+  const user = {
+    id: Math.floor(Math.random() * 1000),
+    name,
+  };
+
+  users.push(user);
+
+  return user;
+};
+
+const updateUser = ({ id, name }) => {
+  const user = getUserById(+id);
+
+  Object.assign(user, { name });
+
+  return user;
+};
+
+const deleteUser = (id) => {
+  users = users.filter((person) => person.id !== +id);
```

Comment:
In the `deleteUser` function, you're not returning anything. It would be a good practice to return a value indicating whether the deletion was successful or not. For example, you could return a boolean or the deleted user object.